### PR TITLE
Implement equals and hashCode on WrappedComponent

### LINKED ISSUE #162: Text Components in Item Components cause ItemStack comparison problems
ItemStacks that should be the exact same cannot be compared to each other using ItemStack#areEqual, and ItemStack#areItemsAndComponets as the item components don't match because of the text created from components in the adventure API.

Both item component maps that are being compared:

{minecraft:custom_name=>net.kyori.adventure.platform.fabric.impl.client.ClientWrappedComponent@44b92b1b, minecraft:lore=>LoreComponent[lines=[net.kyori.adventure.platform.fabric.impl.client.ClientWrappedComponent@ae75c12, empty, net.kyori.adventure.platform.fabric.impl.client.ClientWrappedComponent@7eac21fd, net.kyori.adventure.platform.fabric.impl.client.ClientWrappedComponent@12396f04, empty, net.kyori.adventure.platform.fabric.impl.client.ClientWrappedComponent@1fd57fab, net.kyori.adventure.platform.fabric.impl.client.ClientWrappedComponent@2fe2b78f], styledLines=[literal{Project}[style={color=red,!italic}], empty[style={color=dark_purple,italic}], literal{Information cannot be viewed}[style={color=dark_gray,italic}], literal{from your current node.}[style={color=dark_gray,italic}], empty[style={color=dark_purple,italic}], literal{⇄ Left click to connect}[style={color=blue,!italic}], literal{↔ Right click to cycle}[style={color=blue,!italic}]]], minecraft:max_stack_size=>64, minecraft:enchantments=>ItemEnchantments{enchantments={}, showInTooltip=true}, minecraft:repair_cost=>0, minecraft:attribute_modifiers=>AttributeModifiersComponent[modifiers=[], showInTooltip=true], minecraft:rarity=>EPIC}
 
 
{minecraft:custom_name=>net.kyori.adventure.platform.fabric.impl.client.ClientWrappedComponent@280e45c3, minecraft:lore=>LoreComponent[lines=[net.kyori.adventure.platform.fabric.impl.client.ClientWrappedComponent@35fb131d, empty, net.kyori.adventure.platform.fabric.impl.client.ClientWrappedComponent@76c71fb, net.kyori.adventure.platform.fabric.impl.client.ClientWrappedComponent@17263cec, empty, net.kyori.adventure.platform.fabric.impl.client.ClientWrappedComponent@4d5c4396, net.kyori.adventure.platform.fabric.impl.client.ClientWrappedComponent@20d2d2db], styledLines=[literal{Project}[style={color=red,!italic}], empty[style={color=dark_purple,italic}], literal{Information cannot be viewed}[style={color=dark_gray,italic}], literal{from your current node.}[style={color=dark_gray,italic}], empty[style={color=dark_purple,italic}], literal{⇄ Left click to connect}[style={color=blue,!italic}], literal{↔ Right click to cycle}[style={color=blue,!italic}]]], minecraft:max_stack_size=>64, minecraft:enchantments=>ItemEnchantments{enchantments={}, showInTooltip=true}, minecraft:repair_cost=>0, minecraft:attribute_modifiers=>AttributeModifiersComponent[modifiers=[], showInTooltip=true], minecraft:rarity=>EPIC}

### PR DESCRIPTION
**Pull request by jpenilla**
fixes #162

### CODE REVIEW

File: mod-shared/src/main/java/net/kyori/adventure/platform/modcommon/impl/WrappedComponent.java [hunk 1]
```diff
@@ -166,4 +166,17 @@ public <T> Optional<T> visit(final StyledContentConsumer<T> visitor, final Style
...
   }
+
+  @Override
+  public int hashCode() {
+    return this.deepConverted().hashCode();
+  }
+
+  @Override
+  public boolean equals(final Object obj) {
+    if (obj instanceof WrappedComponent wrappedComponent) {
```
Comment (kashike) [inline mod-shared/src/main/java/net/kyori/adventure/platform/modcommon/impl/WrappedComponent.java]:
```suggestion
    if (obj instanceof final WrappedComponent wrappedComponent) {
```

**Review by kashike — CHANGES_REQUESTED** [review]

**Review by kashike — APPROVED** [review]

**Comment by zml2008** [comment]
Should MC's components' equals() be modified to unwrap WrappedComponnents before comparing?

Otherwise `wrapped.equals(native)` will produce a different result than `native.equals(wrapped)` which is not very fun and cool

**Comment by zml2008** [comment]
I think this is ready to merge, it should be backported to the 5.x series as well though I think since I expect 1.21.1 will live a relatively long life.

**Review by zml2008 — APPROVED** [review]

**Comment by zml2008** [comment]
I've set up `mc/1.21.1` for that backport, I'll do it unless you have the free time.

**Pull request merged**